src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol_ImplementationChecks.cs

```diff
@@ -1261,7 +1261,8 @@ TypeWithAnnotations getNotNullIfNotNullOutputType(TypeWithAnnotations outputType
                     for (var i = 0; i < baseParameters.Length; i++)
                     {
                         var overrideParam = overrideParameters[i + overrideParameterOffset];
-                        if (notNullIfParameterNotNull.Contains(overrideParam.Name) && !baseParameters[i].TypeWithAnnotations.NullableAnnotation.IsAnnotated())
+                        var baseParam = baseParameters[i];
+                        if (notNullIfParameterNotNull.Contains(overrideParam.Name) && NullableWalker.GetParameterState(baseParam.TypeWithAnnotations, baseParam.FlowAnalysisAnnotations).IsNotNull)
```

Comment:
Note that nullable variance checks are shared between method overrides and method group conversions. The delegate type is analogous to a "base" method while the method group is analogous to an "override" method.

Reply:
I think there's a way to observe the impact of the change on an override scenario. Consider adding such tests.